Gifski/Preview/createAVAssetFromGIF.swift

```
@@ -0,0 +1,157 @@
+		AVVideoCodecKey: AVVideoCodecType.h264,
+		AVVideoWidthKey: firstCGImage.width,
+		AVVideoHeightKey: firstCGImage.height
+	])
+	writerInput.expectsMediaDataInRealTime = false
+	let pixelBufferAdaptor = AVAssetWriterInputPixelBufferAdaptor(
+		assetWriterInput: writerInput,
+		sourcePixelBufferAttributes: [
+			kCVPixelBufferPixelFormatTypeKey as String: kCVPixelFormatType_32ARGB,
+			kCVPixelBufferWidthKey as String: firstCGImage.width,
+			kCVPixelBufferHeightKey as String: firstCGImage.height
+		]
+	)
+	guard assetWriter.canAdd(writerInput) else {
+		throw CreateAVAssetError.cannotAddWriterInput
+	}
+	assetWriter.add(writerInput)
+
+	guard assetWriter.startWriting() else {
+		throw CreateAVAssetError.failedToStartWriting
+	}
+	assetWriter.startSession(atSourceTime: .zero)
+
+	let dispatchQueue = DispatchQueue(label: "com.gifski.assetWriterQueue")
+	var frameIndex = 0
+
+	let frameRate: CMTimeScale = await {
+		if let settingFrameRate = settings.conversion.frameRate {
+			return CMTimeScale(settingFrameRate)
+		}
+		if let inputFrameRate = try? await settings.conversion.asset.frameRate {
+			return CMTimeScale(inputFrameRate)
+		}
+		return CMTimeScale(30.0)
+	}()
+
+
+
+	let dataReadyStream = AsyncStream { continuation in
+		writerInput.requestMediaDataWhenReady(on: dispatchQueue) {
+			continuation.yield()
+		}
+	}
+	var progressThreshold = 0.05
+	for await _ in dataReadyStream {
+		while writerInput.isReadyForMoreMediaData && frameIndex < numberOfImages {
+			try Task.checkCancellation()
+			let progress = Double(frameIndex) / Double(numberOfImages)
+			if progress > progressThreshold {
+				onProgress(progress)
+				progressThreshold = progress + 0.05
+			}
+			defer {
+				frameIndex += 1
+			}
+			guard let cgImage = CGImageSourceCreateImageAtIndex(imageSource, frameIndex, nil) else {
+				continue
+			}
+			guard let pixelBufferPool = pixelBufferAdaptor.pixelBufferPool,
+				  let pixelBuffer = createPixelBuffer(from: cgImage, using: pixelBufferPool)
```

Should it really silently skip frames?